Forcibly load our operating_system.rb
This circumvents the GEM_PRELUDE_SUCKAGE by upstream.
Gentoo bug 336863

Upstream: no

diff -Naur 1.9.1-orig/rubygems.rb 1.9.1/rubygems.rb
--- 1.9.1-orig/rubygems.rb	2011-08-24 18:37:34.000000000 +0200
+++ lib/rubygems.rb	2011-08-24 19:01:14.550357134 +0200
@@ -1245,6 +1245,15 @@
   end
 end
 
+begin
+  ##
+  # Force re-loading of operating_system to get Gentoo defaults applied
+
+  load 'rubygems/defaults/operating_system.rb'
+rescue LoadError => e
+  alert_error "Cannot apply Gentoo settings: #{e.message}" 
+end
+
 ##
 # Enables the require hook for RubyGems.
 
